src/Engine/EngineInterface.php

```diff
@@ -10,6 +10,15 @@
 
 interface EngineInterface
 {
+
+    /**
+     * Requires the Analytics instance which handles the Events
+     * that occur during the test process.
+     * 
+     * @param AnalyticsInterface $analytics
+     */
+    public function __construct(AnalyticsInterface $analytics);
```

Comment:
I'm not sure this is a good idea. A constructor shouldn't be part of the interface, it now becomes impossible to create an implementation with a different constructor.